src/xrpld/app/misc/ValidatorList.h

```diff
@@ -343,7 +346,8 @@ class ValidatorList
     load(
         std::optional<PublicKey> const& localSigningKey,
         std::vector<std::string> const& configKeys,
-        std::vector<std::string> const& publisherKeys);
+        std::vector<std::string> const& publisherKeys,
+        std::optional<std::size_t> listThreshold = {});
```

Comment:
Good question, and admittedly I am not 100% certain that this is the right way, so I will present both arguments for and against making a change here (to remove the default)

1. for - in `rippled` proper (as opposed to unit tests), we call this function in one place only `ApplicationImp::setup` and it is important that, in this one location, we definitely **do** provide this parameter, as parsed from the configuration.

2. against - we do recommend using default in configuration, for a good reason. It makes sense to follow this advice, in the most concise way possible (that is, not providing the parameter at all) also in unit tests etc. where we call this function outside of `rippled` proper

3. against - making it a required parameter would cause relatively large churn in unit tests

```
[feature/validator_list_threshold] % git diff --stat
 src/test/app/Manifest_test.cpp          |  2 +-
 src/test/app/ValidatorList_test.cpp     | 66 +++++++++++++++++++++++++++++++++++++-----------------------------
 src/test/app/ValidatorSite_test.cpp     |  2 +-
 src/test/consensus/NegativeUNL_test.cpp |  2 +-
 src/xrpld/app/misc/ValidatorList.h      |  2 +-
 5 files changed, 41 insertions(+), 33 deletions(-)
```

Again, I am not 100% convinced that points 2. and 3. do outweigh point 1. , and (since it's not much work) if you think that 1. (or some other argument) dictates to remove the default value, I'd be happy to do it.

Reply:
I don't have a super strong point of view. Trust your judgement (or defer to any other reviewers)